src/bsoncxx/oid.cpp

```diff
@@ -23,12 +23,7 @@
 namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 
-const oid::init_tag_t oid::init_tag{};
-
-oid::oid() : _is_valid(false) {
-}
-
-oid::oid(init_tag_t) : _is_valid(true) {
+oid::oid() : _is_valid(true) {
```

Comment:
It should throw. Call `bson_oid_is_valid` on the passed in `string_view` and throw (maybe with a new error code) if it is in't valid.